File: app/src/main/java/com/hana053/micropost/interactors/UserMicropostInteractor.java
package com.hana053.micropost.interactors;

import android.support.annotation.Nullable;

import com.hana053.micropost.domain.Micropost;

import java.util.List;

import retrofit2.http.GET;
import retrofit2.http.Path;
import retrofit2.http.Query;
import rx.Observable;

public interface UserMicropostInteractor {

    @GET("users/{userId}/microposts?count=20")
    Observable<List<Micropost>> loadPrevPosts(@Path("userId") long userId, @Nullable @Query("maxId") Long maxId);
}

File: app/src/main/java/com/hana053/micropost/interactors/UserMicropostInteractor.kt
package com.hana053.micropost.interactors

import com.hana053.micropost.domain.Micropost
import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.Query
import rx.Observable


interface UserMicropostInteractor {

    @GET("users/{userId}/microposts?count=20")
    fun loadPrevPosts(@Path("userId") userId: Long, @Query("maxId") maxId: Long?): Observable<List<Micropost>>
}